vpdq/cpp/regtest.py

```diff
@@ -72,91 +74,114 @@ def get_argparse() -> argparse.ArgumentParser:
-                        "-f",
-                        ffmpegPath,
-                        "-r",
-                        secondsPerHash,
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-            else:
-                subprocess.call(
-                    [
-                        "./build/vpdq-hash-video",
-                        "-f",
-                        ffmpegPath,
-                        "-r",
-                        secondsPerHash,
-                        "-d",
-                        outputHashFolder,
-                        "-s",
-                        downsampleFrameDimension,
-                        "-i",
-                        inputVideoFolder + "/" + file,
-                    ]
-                )
-
-    cdir = os.getcwd()
-    pdir = os.path.dirname(cdir)
-    sample = pdir + ("/sample-hashes")
-    output = outputHashFolder
     distanceTolerance = str(args.matchDistanceTolerance)
     qualityTolerance = str(args.qualityTolerance)
-    for file in os.listdir(sample):
-        if file.endswith(".txt"):
-            print("\nMatching File " + file)
-            sampleFile = f"{sample}/{file}"
-            outputFile = f"{output}/{file}"
+    verbose = args.verbose
+
+    if not inputVideoFolder.exists():
+        raise argparse.ArgumentTypeError(
+            f"inputVideoFolder: {inputVideoFolder} does not exist."
+        )
+
+    with TemporaryDirectory() as tempOutputHashFolder:
+        """
+        Write the files to temp
+        If outputHashFolder is specified then
+        copy the output files there, overwriting existing files
+        """
+
+        tempOutputHashFolder = Path(tempOutputHashFolder)
+        if outputHashFolder is not None:
+            if not outputHashFolder.exists():
+                print(f"Creating output directory at {outputHashFolder}")
+                outputHashFolder.mkdir(parents=True, exist_ok=True)
+            print(f"Writing output to directory: {outputHashFolder}")
+        else:
+            print(f"Writing output to temp directory: {tempOutputHashFolder}")
+        # TODO: Add more general options for other video extensions.
+        for fileStr in glob.iglob(f"{inputVideoFolder}/**/*.mp4", recursive=True):
+            file = Path(fileStr)
+
+            """
+            Create output hash file or overwrite existing file
+            This is hardcoded in cpp, and it
+            does not create the file if it does not exist:
+
+            vpdq-hash-video.cpp:
+
+            // Strip containing directory:
+            std::string b = basename(inputVideoFileName, "/");
+            // Strip file extension:
+            b = stripExtension(b, ".");
+            outputHashFileName = outputDirectory + "/" + b + ".txt";
+            """
```

Comment:
blocking: This becomes a string in python, it looks like this wants to be a comment. Use

```
# Create output hash file or overwrite existing file
# This is hardcoded in cpp, and it
# ...
```

Reply:
I'm not sure what the difference is. [PEP257](https://peps.python.org/pep-0257/) states "String literals occurring elsewhere in Python code may also act as documentation." Even [the creator of Python suggests](https://twitter.com/gvanrossum/status/112670605505077248) using these. But I'll change it for style reasons.